randovania/games/prime1/gui/preset_settings/prime_enemy_stat_randomizer.py

```diff
@@ -0,0 +1,188 @@
+        self.range_speed_high.valueChanged.connect(self._on_spin_changed_range_speed_high)
+        self.range_damage_low.valueChanged.connect(self._on_spin_changed_range_damage_low)
+        self.range_damage_high.valueChanged.connect(self._on_spin_changed_range_damage_high)
+        self.range_knockback_low.valueChanged.connect(self._on_spin_changed_range_knockback_low)
+        self.range_knockback_high.valueChanged.connect(self._on_spin_changed_range_knockback_high)
+        self.diff_xyz.toggled.connect(self._on_check_change)
+
+    @classmethod
+    def tab_title(cls) -> str:
+        return "Enemy Attributes"
+
+    @classmethod
+    def uses_patches_tab(cls) -> bool:
+        return True
+
+    def _on_activation(self):
+        checked: bool = self.Activate_Randomizer.isChecked()
+        with self._editor as editor:
+            if checked:
+                editor.set_configuration_field("enemy_attributes", EnemyAttributeRandomizer(self.range_scale_low.value(), self.range_scale_high.value(),
+                                                                                            self.range_health_low.value(), self.range_health_high.value(),
+                                                                                            self.range_speed_low.value(), self.range_speed_high.value(),
+                                                                                            self.range_damage_low.value(), self.range_damage_high.value(),
+                                                                                            self.range_knockback_low.value(), self.range_knockback_high.value(),
+                                                                                            self.diff_xyz.isChecked()))
+            else:
+                editor.set_configuration_field("enemy_attributes", None)
+        self.minimum_label.setEnabled(checked)
+        self.maximum_label.setEnabled(checked)
+        self.scale_attribute_label.setEnabled(checked)
+        self.range_scale_low.setEnabled(checked)
+        self.range_scale_high.setEnabled(checked)
+        self.health_attribute_label.setEnabled(checked)
+        self.range_health_low.setEnabled(checked)
+        self.range_health_high.setEnabled(checked)
+        self.speed_attribute_label.setEnabled(checked)
+        self.range_speed_low.setEnabled(checked)
+        self.range_speed_high.setEnabled(checked)
+        self.damage_attribute_label.setEnabled(checked)
+        self.range_damage_low.setEnabled(checked)
+        self.range_damage_high.setEnabled(checked)
+        self.knockback_attribute_label.setEnabled(checked)
+        self.range_knockback_low.setEnabled(checked)
+        self.range_knockback_high.setEnabled(checked)
+        self.diff_xyz.setEnabled(checked)
+        self.label.setEnabled(checked)
+        self.label_2.setEnabled(checked)
+
+    def on_preset_changed(self, preset: Preset):
+        config = preset.configuration
```

Comment:
You should set the state of `self.Activate_Randomizer` accordingly here too

Reply:
You're missing setting to `False` when enemy_attributes  is None

Reply:
Thats weird, I did that in my copy but it didn't commit that....but it commited mostly everything else?

Reply:
You're still missing a `self.activate_randomizer.setChecked(False)` in the else case